### ISSUE TITLE
Systemjs initial load performance

### ISSUE THREAD

**Issue by m-j** [systemjs/systemjs#1325]
I'm struggling with systemjs initial load performance. On solid desktop machine it is acceptable, but on mobile it is way too high. I don't care too much about resource load time as bundling does good job here for http 1. The problem seems to be execution time of sfx bundle. I'm using systemjs with jspm and I bundle app using `jspm.bundleSFX()` which uses `systemBuilder.buildStatic()` underneath.

Now let's establish what I'm measuring. All measurements are done in Chrome, results are averages of few page refreshes. I'm measuring app.min.js execution time by using timeline:
https://drive.google.com/file/d/0B3MBFJcy6DPnQVprMW94WGlyRDg/view?usp=sharing

And also execution of first line app.min.js:
https://drive.google.com/file/d/0B3MBFJcy6DPnZkl2cTE3cW56TTg/view?usp=sharing which as I believe it's code responsible for module registration and dependency resolution (but that's my guess as I didn't want to read minified code).

Below are results for decent android tablet and pretty good PC.

```
Nvidia shield K1 tablet       |                                 |i7-490K 4Ghz 16GB RAM         |                                 
app.min.js execution time [ms]|Module loader execution time [ms]|app.min.js execution time [ms]|Module loader execution time [ms]
1616.666667                   |1031                             |396.3333333                   |234.3333333                      
  
```

Is there any way to produce plain simple static bundle with systemjs similar to one that for example r.js produces with no magical loader at the beginning just es5 code?

**Comment by jakeNiemiec** [comment]
Short answer: [bundle the things what won’t change much, and separate the things that change often.](https://github.com/jspm/jspm-cli/blob/0.17/docs/production-workflows.md#creating-a-bundle-with-arithmetic) 

Soup answer: Imagine you have 2 barrels, one empty and filled with different vegetables. Now, you love cabbage, you eat it almost every day. You have a variety of different kinds of cabbages. Everyone knows that you don't eat an entire cabbage for a meal. When you go to grab some, you find your self digging around the other veggies to find the perfect piece of cabbage, cut a portion off, and toss it back in. 

##### Now, it goes without saying, that you can save some time by having a dedicated barrel for all cabbages:
`jspm bundle soupRecipe.js - cabbages.js soupWithoutTheCabbage.js --inject`
^bundle everything according to `soupRecipe.js` into `soupWithoutTheCabbage.js` except the cabbages because those will change often. `soupRecipe.js` includes dependancies such as `hotWater.js`, `jSpice` and `NaCl`, things that are in most soups and will not change often. In this way, the rest is done for you.

##### Or you may decide to prepare all of it beforehand:
`jspm build soupRecipe.js cabbageSoup.js`
Cabbage soup mix. You can't reach in and take out the cabbages, if you want to change it, you have to start from the beginning. 

##### Or you may want to store your ingredients in the exact order you need them:
`jspm depcache soupRecipe.js -o allIngredientsInOrder.js`
All of the ingredients are labeled 1, 2, 3..., you can add them quickly without thinking.

In your case you are rooting through a barrel, following arbitrary instructions. Worse than that, you store all of your vegetables in the shed out back, and you can only carry 3 ingredients at a time to your kitchen (http requests with your web browser).

**Comment by jakeNiemiec** [comment]
>Is there any way to produce plain simple static bundle with systemjs similar to one that for example r.js produces with no magical loader at the beginning just es5 code?

tl;dr: Learn to use [`jspm build`](http://jspm.io/0.17-beta-guide/static-builds-with-rollup-optimization.html)
e.g. `jspm build mySrc  dist/myBuildApp.js` (add `--format global` if that is what you were intending in your screenshot).

**Comment by guybedford** [comment]
The core sfx loader is just like an almond-style loader, the only thing extra it provides is the ability to know how to link es modules alongside commonjs modules and the circular reference handling for that case.

If you write all modules as es modules, then it leaves this loader header out.

I believe sfx bundles are fairly well optimized though - how large is this build file in the first place?
